api/src/main/java/com/datatorrent/api/Context.java

```diff
@@ -301,6 +301,12 @@
      */
     int getId();
 
+    /**
+     * Return the number of windows before the next checkpoint including the current window.
+     * @return Number of windows from checkpoint, 1 if the checkpoint will be after the current window
+     */
+    int getWindowsFromCheckpoint();
```

Comment:
@ilooner the implementation for this context is in the engine itself so the exposure is contained. There are one or two implementations in the malhar tests but those won't affect the runtime of the app. If we still think this is a problem this can be made an attribute.